etc/run-clang-tidy.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
```

Comment:
To confirm, developers should be able to run this themselves, `./etc/run-clang-tidy.sh`, right? If so, I'd make a note in https://github.com/mongodb/mongo-cxx-driver/blob/master/CONTRIBUTING.md to run this script before submitting a PR.